### PR TITLE
Math extensions update to include bitwise ops and floating point helpers

### PR DESCRIPTION
**Pull request by TristonianJones**
Support for floating point helpers and bitwise operators.

### CODE REVIEW

File: ext/math_test.go [hunk 1]
```diff
@@ -112,6 +112,74 @@ func TestMath(t *testing.T) {
 				"numbers": []float64{-21.0, -10.5, 1.0},
 			},
 		},
+
+		// Tests for math bitwise operators
+		// Signed bitwise ops
+		{expr: "math.bitAnd(1, 2) == 0"},
```
Comment (l46kok) [inline ext/math_test.go:118]: Up to you but probably worth adding some tests around mixed int/uint literals, for no matching overload and ones wrapped in `dyn` to demonstrate that it fails during eval

File: ext/math.go [hunk 2]
```diff
@@ -86,28 +87,311 @@ import (
...
 //	math.least(dyn('string')) // runtime error
+//
+// # Math.BitOr
+//
+// Introduced at version: 1
+//
+// Var-args macro which accepts a variable quantity of [1, 32) integers and
+// performs a bitwise-OR operation on all of them. The macro is expanded into
+// a call to the internal-only overload of math.@bitOr(number),
+// math.@bitOr(number, number), or math.@bitOr(list)
```
Comment (l46kok) [inline ext/math.go]: Is vararg macro expansion and the list overload coming in follow-up PRs?

File: ext/README.md [hunk 3]
```diff
@@ -117,6 +118,249 @@ Examples:
...
+    math.bitAnd(-3, -5)   // returns -7
+
+### Math.BitXor
+
+Introduced at version: 1
+
+    math.bitXor(<int>, <int>) -> <int>
+    math.bitXor(<uint>, <uint>) -> <uint>
+
+Performs a bitwise-XOR  operation over two int or uint values.
```
Comment (l46kok) [inline ext/README.md]:
```suggestion
Performs a bitwise-XOR operation over two int or uint values.
```

File: ext/math.go [hunk 4]
```diff
@@ -86,28 +87,311 @@ import (
...
+//
+//	math.abs(<double>) -> <double>
+//	math.abs(<int>) -> <int>
+//	math.abs(<uint>) -> <uint>
+//
+// Examples:
+//
+//	math.abs(-1)  // returns 1
+//	math.abs(1)   // returns 1
+//	math.abs(-9223372036854775808) // overlflow error
```
Comment (l46kok) [inline ext/math.go]:
```suggestion
//	math.abs(-9223372036854775808) // overflow error
```

File: ext/math_test.go [hunk 5]
```diff
@@ -319,11 +387,32 @@ func TestMathRuntimeErrors(t *testing.T) {
...
+		{
+			expr: "math.abs(-9223372036854775808)",
+			err:  "overflow",
+		},
 	}
 
 	env := testMathEnv(t,
 		cel.Variable("a", cel.DynType),
 		cel.Variable("b", cel.IntType),
+		cel.Variable("ints", cel.ListType(cel.IntType)),
```
Comment (l46kok) [inline ext/math_test.go]: Unused, but maybe you meant to include overloads for lists

File: ext/math.go [hunk 6]
```diff
@@ -86,28 +87,311 @@ import (
...
+//
+// Perform a left shift of bits on the first parameter, by the amount of bits
+// specified in the second parameter. The first parameter is either a uint or
+// an int. The second parameter must be an int.
+//
+// When the second parameter is 64 or greater, 0 will be always be returned
+// since the number of bits shifted is greater than or equal to the total bit
+// length of the number being shifted.
+//
+//	math.bitShiftLeft(<int>, <int>) -> <int>
```
Comment (l46kok) [inline ext/math.go:163]: Should we just define the second parameter to be uint? In most languages, negative shift is either undefined or an error.

File: ext/README.md [hunk 7]
```diff
@@ -117,6 +118,249 @@ Examples:
...
+
+    math.abs(<double>) -> <double>
+    math.abs(<int>) -> <int>
+    math.abs(<uint>) -> <uint>
+
+Examples:
+
+    math.abs(-1)  // returns 1
+    math.abs(1)   // returns 1
+    math.abs(-9223372036854775808) // overlflow error
```
Comment (l46kok) [inline ext/README.md:298]:
```suggestion
    math.abs(-9223372036854775808) // overflow error
```

File: ext/math.go (hunk 6 above)
Comment (TristonianJones) [reply ext/math.go:163]: Within ZetaSQL, the shift argument is an int. I kept it for parity there.

File: ext/math_test.go (hunk 5 above)
Comment (TristonianJones) [reply ext/math_test.go]: Thanks, it was from a prior test involving the vararg macros

File: ext/math.go (hunk 2 above)
Comment (TristonianJones) [reply ext/math.go]: I had a var-arg macro and then removed it. The comment was extant.

**Comment by TristonianJones** [comment]
PTAL

File: ext/math.go (hunk 6 above)
Comment (l46kok) [reply ext/math.go:163]: We should document this behavior, this will have to be special cased in both Java/C++ to produce an error

**Review by l46kok — DISMISSED** [review]

File: ext/math.go (hunk 6 above)
Comment (TristonianJones) [reply ext/math.go:163]: Oh, I see you mean documenting the error case for a negative shift?

Comment (l46kok) [reply ext/math.go:163]: Yep, negative shifts isn't clearly established across most languages (It either results in an error, always returns 0, shift direction is flipped, with ambiguity introduced between logical/arithmetic shifts etc.).

Comment (TristonianJones) [reply ext/math.go:163]: Done. I'll add conformance tests as well.

**Review by l46kok — APPROVED** [review]

**Pull request merged**